python/ray/rllib/models/preprocessors.py

```diff
@@ -140,6 +140,9 @@ def _init_shape(self, obs_space, options):
 
     @override(Preprocessor)
     def transform(self, observation):
+        if not self._obs_space.contains(observation):
+            raise ValueError("Observation outside expected value range",
+                             self._obs_space, observation)
```

Comment:
Can you also call this in the other preprocessors (maybe this can be moved into the Preprocessor base class as a method and called by subclasses?)